### PR TITLE
add port check code for mesos dispatcher

### PR DESCRIPTION
**Pull request by skonto**
- check port util code
- check if mesos dispather is up. Sometimes dispatcher didnt come up. when run the tests multiple times against the same cluster. Next thing to check is killing the dispatcher etc.

### CODE REVIEW

**Comment by dragos** [comment]
This is pretty old. Any reason why nobody had a look yet?

**Comment by dragos** [comment]
I'll do it now :)

File: test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/Utils.scala [hunk 1]
```diff
@@ -132,3 +139,63 @@ object Utils {
     Process(s"${sparkHome}/sbin/stop-mesos-dispatcher.sh").!
   }
 }
+
+object ServiceUtil {
+
+
+  private def checkAvailablePort(host:String, port:Int): Boolean = {
```
Comment (dragos) [inline test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/Utils.scala]: Nitpick: formatting (have a space between `:` and type)

File: test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/Utils.scala [hunk 2]
```diff
@@ -132,3 +139,63 @@ object Utils {
...
+
+    var res = false
+    var p : Option[Socket] = None
+
+    try{
```
Comment (dragos) [inline test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/Utils.scala]: formatting

File: test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/Utils.scala [hunk 3]
```diff
@@ -132,3 +139,63 @@ object Utils {
...
+    }
+
+    res
+  }
+
+  private def waitOnCheck[T] (service: String, delay: Int, iteration: Int, max:Int)(body: => Boolean): Boolean = {
+
+    val rs = body
+    println(s"Checking for service $service - try $iteration...")
+    if(!rs){
```
Comment (dragos) [inline test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/Utils.scala]: formatting

File: test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/Utils.scala [hunk 4]
```diff
@@ -132,3 +139,63 @@ object Utils {
...
+          p.get.close()
+        } catch {
+          case e: Exception => e.printStackTrace()   //closing stream failed, print stack
+        }
+
+      }
+
+    }
+
+    res
```
Comment (dragos) [inline test-runner/src/main/scala/com/typesafe/spark/test/mesos/framework/runners/Utils.scala]: Please remove all these empty lines

**Comment by dragos** [comment]
Except for formatting, it looks good to me. I know this is annoying, but let's try to keep a consistent feel in the code base. A good start is the [official Scala code style](http://docs.scala-lang.org/style/)

**Comment by skonto** [comment]
i have used http://www.scalastyle.org/ in the past which is enforces style and found it really good for code quality.

**Comment by dragos** [comment]
Lovely! LGTM

**Pull request merged**